Include/internal/pycore_long.h

```diff
@@ -110,6 +110,15 @@ PyAPI_FUNC(char*) _PyLong_FormatBytesWriter(
     int base,
     int alternate);
 
+/* Return 1 if the argument is positive single digit int */
+static inline int
+_PyLong_IsPositiveSingleDigit(PyObject* sub) {
+    // this method uses the twos-complement representation
```

Comment:
This comment feels disconnected (two's complement is not unique to this function, it is assumed everywhere).

What requires an explanation is the trick of casting a signed value to an unsigned value and then checking whether the result is <= 1. The clever bit here is that this cast makes _all_ negative numbers be considered very large positive numbers.

I'm also not keen on 'signed_magnitude' as the name for the variable. It makes me think of the "sign + magnitude" representation of numbers which is actually how I'd describe one's complement (!). I suggest renaming it to 'signed_size' which is just a reminder of what Py_SIZE() of a PyLong represents the size, negated if the sign of the overall number is negative. (The clever bit there is that the value zero has size 0 which is invariant if negated.)

Reply:
What's a cast from int to uint expected to do for negative values in one's complement? Thinking about it more, it probably still converts all negative numbers to very large positive ones, so it would still work, except for -0. Or maybe even in that case, because that's not a _positive_ int.

So I'm still not sure that two's complement deserves being called out here.

(I do agree that we should stop believing we might support one's complement. :-)

Reply:
I agree that mentioning twos complement here is not relevant. I updated the description and included the link that was already present in the [valid_index](https://github.com/python/cpython/blob/d5f8a2b6ad408368e728a389da918cead3ef7ee9/Objects/listobject.c#L237) method.

I also updated the name to `signed_size`.